db/system_keyspace.cc

```diff
@@ -288,6 +288,26 @@ schema_ptr system_keyspace::raft_config() {
     return schema;
 }
 
+schema_ptr system_keyspace::repair_history() {
+    static thread_local auto schema = [] {
+        auto id = generate_legacy_id(NAME, REPAIR_HISTORY);
+        return schema_builder(NAME, REPAIR_HISTORY, std::optional(id))
+            .with_column("table_uuid", uuid_type, column_kind::partition_key)
+            // The time is repair start time
+            .with_column("repair_time", timestamp_type, column_kind::clustering_key)
+            .with_column("repair_uuid", uuid_type, column_kind::clustering_key)
+            // The token range is (range_start, range_end]
+            .with_column("range_start", long_type, column_kind::clustering_key)
+            .with_column("range_end", long_type, column_kind::clustering_key)
```

Comment:
Let's say a subrange was moved out by bootstrap and brought back in be decommission. Then we will have a repair entry valid for that subrange, even though it wasn't under this node's control for some time (after the repair_time column).

I don't think it's a problem. Is it?

Reply:
It is not a problem. The valid entry for that sub range would not cover what happens after the repair. We need a new repair for the sub range on this node to gc the new data.

Reply:
Won't we have an older entry in the table? Suppose the new owner (that later was decommissioned) didn't repair?

Reply:
For example, we have an old entry saying range (1, 100] is repaired at t1. After (1,100] was moved out an then brought back again at t2, this entry is still valid for gc because moving the range out and in does not change the fact the range (1, 100] was repaired at t1.